crypten/mpc/primitives/arithmetic.py

```diff
@@ -416,12 +416,7 @@ def div_(self, y):
 
             # Truncate protocol for dividing by public integers:
             if comm.get().get_world_size() > 2:
-                wraps = self.wraps()
-                self.share //= y
-                # NOTE: The multiplication here must be split into two parts
-                # to avoid long out-of-bounds when y <= 2 since (2 ** 63) is
-                # larger than the largest long integer.
-                self -= wraps * 4 * (int(2 ** 62) // y)
+                self.mul_(1/y)
```

Comment:
I don't quite understand how this would work. Essentially, every multiply (except by an integer) requires multiplication, then truncation by the encoding scale (usually 2 ** 16), which will cause an infinite loop here.

Am I missing something?

Reply:
The original idea is here is to turn public division into multiplication of its reciprocal, then the `truncation` function in the multiplication protocol will act as the replacement of `wrap` function. But it seems like this is causing some infinite loop in certain cases. I will reexamine this and see if I could find a better way of handling it.